### PR TITLE
[bugfix] reset schema_template_external_epg_subnet name attr to fix terraform import

### LINKED ISSUE #112: Minor issue - importing mso_schema_template_external_epg_subnet sets a name value of jsonencode({})
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

terraform version
Terraform v1.1.4
on windows_amd64

provider registry.terraform.io/ciscodevnet/mso v0.4.1

### MSO version

Nexus Dashboard
Version 2.0(2g)

Multi-Site Orchestrator
Version: 3.3(1e)

dCloud Cisco Nexus Dashboard Orchestrator for ACI Lab v1

### APIC version and APIC Platform for Site Level Resources

N/A

### Affected Resource(s)

* mso_schema_template_external_epg_subnet

### Terraform Configuration Files
```hcl
terraform {
  required_providers {
    mso = {
      source  = "CiscoDevNet/mso"
      version = "0.4.1"
    }
  }
}

provider "mso" {
  # Creds provided as environment variables
}

resource "mso_schema_template_external_epg_subnet" "eepg_subnet" {
  aggregate         = []
  external_epg_name = "test_eepg"
  ip                = "10.0.0.0/8"
  schema_id         = "62165ca82e0000150370b5de"
  scope = [
    "export-rtctrl",
    "import-security",
  ]
  template_name = "test_template"
}

```
### Expected Behavior
Manually created configuration should not include the value jsonencode({}) for an attribute that is not exposed in the MSO GUI.

Whilst this is a minor issue that has no serious impact, the value should be returned as "null" or the value omitted from terraform configuration altogether. 

### Actual Behavior
If an existing external epg has been created in the MSO GUI, importing it causes the name value to be jsonencode({}). This can then not be omitted in the Terraform configuration. It must either be specified exactly as jsonencode({}) or replaced with an alternative value.

This name attribute is not exposed in the GUI.

### Steps to Reproduce
1. Create an external EEPG with a subnet associated in the MSO GUI under a schema template
2. Import the resource into Terraform

#### GUI Configuration
image

#### Import command and output
```hcl
terraform import 'mso_schema_template_external_epg_subnet.eepg_subnet' '62165ca82e0000150370b5de/tempalte/test_template/externalEPG/test_eepg/ip/10.0.0.0/8'
mso_schema_template_external_epg_subnet.eepg_subnet: Importing from ID "62165ca82e0000150370b5de/tempalte/test_template/externalEPG/test_eepg/ip/10.0.0.0/8"...
mso_schema_template_external_epg_subnet.eepg_subnet: Import prepared!
  Prepared mso_schema_template_external_epg_subnet for import
mso_schema_template_external_epg_subnet.eepg_subnet: Refreshing state... [id=10.0.0.0]

Import successful!

The resources that were imported are shown above. These resources are now in
your Terraform state and will henceforth be managed by Terraform.

```

#### Terraform state after import
```hcl
$ terraform show
# mso_schema_template_external_epg_subnet.eepg_subnet:
resource "mso_schema_template_external_epg_subnet" "eepg_subnet" {
    aggregate         = []
    external_epg_name = "test_eepg"
    id                = "10.0.0.0/8"
    ip                = "10.0.0.0/8"
    name              = jsonencode({})
    schema_id         = "62165ca82e0000150370b5de"
    scope             = [
        "export-rtctrl",
        "import-security",
    ]
    template_name     = "test_template"
}
```

#### Apply of configuration with matches the GUI (except for the non-existent name attribute)
```hcl
$ terraform plan
mso_schema_template_external_epg_subnet.eepg_subnet: Refreshing state... [id=10.0.0.0/8]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # mso_schema_template_external_epg_subnet.eepg_subnet will be updated in-place
  ~ resource "mso_schema_template_external_epg_subnet" "eepg_subnet" {
        id                = "10.0.0.0/8"
      - name              = jsonencode({}) -> null
        # (6 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### PR DESCRIPTION
**Pull request by xinyuezhao**
Set external epg subnet ip as ForceNew
Set external epg subnet name as empty string in resourceMSOTemplateExtenalepgSubnetImport()
resolve #112

### CODE REVIEW

**Comment by xinyuezhao** [comment]
add a description indicatiing that 'name' won't be showed in UI after subnet being created

File: mso/datasource_mso_schema_template_external_epg_subnet.go [hunk 1]
```diff
@@ -41,11 +41,13 @@ func dataSourceMSOTemplateExternalEpgSubnet() *schema.Resource {
 				Type:         schema.TypeString,
 				Required:     true,
 				ValidateFunc: validation.StringLenBetween(1, 1000),
+				ForceNew:     true,
 			},
 			"name": &schema.Schema{
 				Type:         schema.TypeString,
 				Optional:     true,
 				ValidateFunc: validation.StringLenBetween(1, 1000),
+				Default:      nil,
```
Comment (lhercot) [inline mso/datasource_mso_schema_template_external_epg_subnet.go:50]: Do we need this default or should we have a else condition later?

File: mso/resource_mso_schema_template_external_epg_subnet.go [hunk 2]
```diff
@@ -50,11 +50,13 @@ func resourceMSOTemplateExtenalepgSubnet() *schema.Resource {
 				Type:         schema.TypeString,
 				Required:     true,
 				ValidateFunc: validation.StringLenBetween(1, 1000),
+				ForceNew:     true,
 			},
 			"name": &schema.Schema{
 				Type:         schema.TypeString,
 				Optional:     true,
 				ValidateFunc: validation.StringLenBetween(1, 1000),
+				Default:      nil,
```
Comment (lhercot) [inline mso/resource_mso_schema_template_external_epg_subnet.go:59]: Do we need this default or should we have a else condition later?

File: website/docs/r/schema_template_external_epg_subnet.html.markdown [hunk 3]
```diff
@@ -30,7 +30,7 @@ resource "mso_schema_template_external_epg_subnet" "subnet1" {
 * `template_name` - (Required) Template where External EPG Subnet to be created.
 * `external_epg_name` - (Required) Name of External EPG.
 * `ip` - (Required) The IP range in CIDR notation.
-* `name` - (Optional) Name of Subnet.
+* `name` - (Optional) Name of Subnet. Name can be set through this module but it is not exposed in UI.
```
Comment (lhercot) [inline website/docs/r/schema_template_external_epg_subnet.html.markdown:33]: through this resource but... in the UI

**Review by lhercot — CHANGES_REQUESTED** [review]

File: mso/resource_mso_schema_template_external_epg_subnet.go (hunk 2 above)
Comment (xinyuezhao) [reply mso/resource_mso_schema_template_external_epg_subnet.go]: added a condition in mso-go-client to check whether Name attr exist

File: website/docs/r/schema_template_external_epg_subnet.html.markdown (hunk 3 above)
Comment (lhercot) [reply website/docs/r/schema_template_external_epg_subnet.html.markdown]: Second part "in the UI" not fixed yet.

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**